internal/provider/resource_gitlab_project_issue_test.go

```diff
@@ -0,0 +1,230 @@
+					testAccCheckGitlabProjectIssueExists("gitlab_project_issue.this", &testIssue),
+					resource.TestCheckResourceAttr("gitlab_project_issue.this", "project", testProject.PathWithNamespace),
+					resource.TestCheckResourceAttr("gitlab_project_issue.this", "iid", "1"),
+					resource.TestCheckResourceAttr("gitlab_project_issue.this", "title", "Terraform test issue"),
+					resource.TestCheckResourceAttr("gitlab_project_issue.this", "assignee_ids.#", "0"),
+					resource.TestCheckResourceAttr("gitlab_project_issue.this", "confidential", "false"),
+					resource.TestCheckResourceAttr("gitlab_project_issue.this", "description", ""),
+					resource.TestCheckResourceAttr("gitlab_project_issue.this", "labels.#", "0"),
+					resource.TestCheckResourceAttr("gitlab_project_issue.this", "issue_type", "issue"),
+				),
+			},
+			// close issue
+			{
+				Config: testAccGitlabProjectIssueConfigWithState(testProject, "closed"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGitlabProjectIssueExists("gitlab_project_issue.this", &testIssue),
+					resource.TestCheckResourceAttr("gitlab_project_issue.this", "state", "closed"),
+				),
+			},
+			// re-open issue
+			{
+				Config: testAccGitlabProjectIssueConfigWithState(testProject, "opened"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGitlabProjectIssueExists("gitlab_project_issue.this", &testIssue),
+					resource.TestCheckResourceAttr("gitlab_project_issue.this", "state", "opened"),
+				),
+			},
+			// Verify import
+			{
+				ResourceName:            "gitlab_project_issue.this",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"delete_on_destroy"},
+			},
+		},
+	})
+}
+
+func TestAccGitlabProjectIssue_deleteOnDestroy(t *testing.T) {
+	testAccCheck(t)
+
+	testProject := testAccCreateProject(t)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: providerFactories,
+		CheckDestroy:      testAccCheckGitlabProjectIssueDestroy,
+		Steps: []resource.TestStep{
+			// create Issue with required values only
+			{
+				Config: testAccGitlabProjectIssueConfigDeleteOnDestroy(testProject),
+			},
+		},
+	})
+}
+
+func testAccCheckGitlabProjectIssueExists(n string, issue *gitlab.Issue) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		project, issueIID, err := resourceGitLabProjectIssueParseId(rs.Primary.ID)
+		if err != nil {
+			return fmt.Errorf("Error parsing issue ID: %s", err)
+		}
+
+		gotIssue, _, err := testGitlabClient.Issues.GetIssue(project, issueIID)
+		if err != nil {
+			return fmt.Errorf("Cannot get issue: %v", err)
+		}
+
+		log.Printf("[DEBUG] testAccCheckGitlabProjectIssueExists: %#v", gotIssue)
+
+		*issue = *gotIssue
+		return nil
+	}
+}
+
+func testAccCheckGitlabProjectIssueDestroyAsClosed(s *terraform.State) error {
```

Comment:
You could rewrite this as a single function for both `testAccCheckGitlabProjectIssueDestroyAsClosed` and `testAccCheckGitlabProjectIssueDestroy`, and alter your assertion based on the `delete_on_destroy` value. IMO this would be less error prone for test refactors.